packages/build/src/install/missing.js

```diff
@@ -82,7 +82,7 @@ const getIntegrationPackage = async function ({ integration: { version, dev }, c
         throw new Error(res.stdout)
       }
     } catch (e) {
-      throw new Error(`Failed to build integration`)
+      throw new Error(`Failed to build integration: ${e.message ? e.message : e}`)
```

Comment:
it seems like `e.message` will most likely be multi-line output from `npm run build`- does that work okay with the custom error parsing stuff that `@netlify/build` has going on?

looking at that error handling I also wonder if a custom ErrorType would be useful here- it looks like `stackType: 'message'` kind of does what we're attempting to do here, by using the other process' output as the error stack

Reply:
@cubeghost can you explain what you mean with the custom error parsing stuff? :O I don't think I'm aware of that

Reply:
yeah! I noticed a bunch of code in the `src/error` directory, particularly [`src/error/types.ts`](https://github.com/netlify/build/blob/main/packages/build/src/error/types.ts#L52-L68), and looked at the corresponding [documentation](https://github.com/netlify/build/blob/main/packages/build/docs/errors.md). this code not only customizes the error output but also does a bunch of parsing and formatting of stack traces and such.

from what I can see from [Uma's error](https://netlify.slack.com/archives/C043UGRAC3F/p1709930455452129?thread_ts=1709308717.515559&cid=C043UGRAC3F), this error ends up unhandled so it follows the generic `exception` [rules](https://github.com/netlify/build/blob/main/packages/build/src/error/types.ts#L507-L513), and it ends up logging two stacks, one for where we're throwing inside of `@netlify/build` and a less visible one for the actual error in the underlying process. 

so I think if we added a new error type specifically for integration build failures, we could customize where the error stack comes from, as well as customizing the error message